src/MSALWrapper/AuthFlow/AuthFlowFactory.cs

```diff
@@ -34,9 +35,18 @@ public class AuthFlowFactory
             string preferredDomain,
             string promptHint,
             string osxKeyChainSuffix,
-            IPCAWrapper pcaWrapper = null)
+            IPCAWrapper pcaWrapper = null,
+            IPlatformUtils platformUtils = null)
         {
+            logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            platformUtils = platformUtils ?? new PlatformUtils(logger);
             List<IAuthFlow> flows = new List<IAuthFlow>();

```

Comment:
Might be worth adding a small comment here that the _order_ in which flows are added to this list is **very** important.